R/plotCCA.R

```diff
@@ -0,0 +1,540 @@
+                vector_label <- .tidy_vector_labels(vector_label, coldata, ...)
+            }
+            # Add to df
+            vector_data$vector_label <- vector_label
+        } else{
+            # Check that user-provided labels are correct length
+            if( length(vec_lab) != nrow(vector_data) ){
+                stop("Number of labels in 'vec_lab' do not match with number of vectors.",
+                     call. = FALSE)
+            }
+            # If they are, add labels to data
+            vector_data$vector_label <- vec_lab
+        }
+    }
+    
+    # Get significance data
+    signif_data <- NULL
+    if( add_significance && !is.null(vector_data) && all_var_found ){
+        # Check if data is available
+        ind <- names(attributes(reduced_dim)) %in% c("significance")
+        # If it can be found
+        if( any(ind) ){
+            # Get biplot from cca object
+            signif_data <- attributes(reduced_dim)[ind][[1]]
+            signif_data <- signif_data[["permanova"]]
+            signif_data <- as.data.frame(signif_data)
+            # Add significance to vector labels
+            # Get vector labels
+            vector_label <- vector_data[["vector_label"]]
+            # Add significance to vector labels
+            vector_label <- .add_signif_to_vector_labels(vector_label, variable_names, signif_data, ...)
+            vector_data[["vector_label"]] <- vector_label
+        } else{
+            # If it cannot be found, give warning
+            warning("Significance data was not found. please calculate CCA by using runCCA.",
+                    call. = FALSE)
+        }
+    }
+    
+    # Create labels for axis
+    xlab <- paste0(dimred, " 1")
+    ylab <- paste0(dimred, " 2")
+    if( add_expl_var ){
+        # Check if data is available
+        ind <- names(attributes(reduced_dim)) %in% c("rda", "cca")
+        # If it can be found
+        if( any(ind) ){
+            # Add explained variance
+            rda <- attributes(reduced_dim)[ind][[1]]
+            xlab <- paste0(
+                xlab, " (",
+                format(round( summary(rda)$concont$importance[2, 1]*100, 1 ), nsmall = 1 ), "%)")
+            ylab <- paste0(
+                ylab, " (",
+                format(round( summary(rda)$concont$importance[2, 2]*100, 1 ), nsmall = 1 ), "%)")
+        } else{
+            # If it cannot be found, give warning
+            warning("CCA object was not found. Please calculate CCA by using runCCA.", call. = FALSE)
+        }
+    }
+    
+    # Create a list to return
+    result <- list(
+        plot = plot,
+        ellipse_data = ellipse_data,
+        vector_data = vector_data,
+        xlab = xlab,
+        ylab = ylab
+    )
+    return(result)
+}
+
+# Make vector labels more tidy, i.e, separate variable and group names.
+# Replace also underscores with space
+.tidy_vector_labels <- function(
+        vector_label, coldata, sep_group = "\U2012", sep_underscore = " ", ...){
```

Comment:
This is just internal stuff, but you could still add input checks for `sep_group `and `sep_underscore`. Better name for sep_underscore would be repl_underscore since underscore is replaced